src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -209,4 +224,143 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     {
         e.Handled(true);
     }
+
+    // Method Description:
+    // - Handler for changing the text. Triggers SearchChanged event
+    // Arguments:
+    // - sender: not used
+    // - e: event data
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::TextBoxTextChanged(winrt::Windows::Foundation::IInspectable const& /*sender*/, winrt::Windows::UI::Xaml::RoutedEventArgs const& /*e*/)
+    {
+        _SearchChangedHandlers(TextBox().Text(), _GoForward(), _CaseSensitive());
+    }
+
+    // Method Description:
+    // - Handler for clicking the case sensitivity toggle. Triggers SearchChanged event
+    // Arguments:
+    // - sender: not used
+    // - e: not used
+    // Return Value:
+    // - <none>
+    void SearchBoxControl::CaseSensitivityButtonClicked(winrt::Windows::Foundation::IInspectable const& /*sender*/, winrt::Windows::UI::Xaml::RoutedEventArgs const& /*e*/)
+    {
+        _SearchChangedHandlers(TextBox().Text(), _GoForward(), _CaseSensitive());
+    }
+
+    // Method Description:
+    // - Formats a status message representing the search state:
+    // * "Searching" - if totalMatches is negative
+    // * "No results" - if totalMatches is 0
+    // * "?/n" - if totalMatches=n matches and we didn't start the iteration over matches
+    // (usually we will get this after buffer update)
+    // * "m/n" - if we are currently at match m out of n.
+    // * "m/max+" - if n > max results to show
+    // * "?/max+" - if m > max results to show
+    // Arguments:
+    // - totalMatches - total number of matches (search results)
+    // - currentMatch - the index of the current match (0-based)
+    // Return Value:
+    // - status message
+    winrt::hstring SearchBoxControl::_FormatStatus(int32_t totalMatches, int32_t currentMatch)
+    {
+        if (totalMatches < 0)
+        {
+            return RS_(L"TermControl_Searching");
+        }
+
+        if (totalMatches == 0)
+        {
+            return RS_(L"TermControl_NoMatch");
+        }
+        std::wstringstream ss;
+
+        if (currentMatch < 0 || currentMatch > (MaximumTotalResultsToShowInStatus - 1))
+        {
+            ss << CurrentIndexTooHighStatus;
+        }
+        else
+        {
+            ss << currentMatch + 1;
+        }
+
+        ss << StatusDelimiter;
+
+        if (totalMatches > MaximumTotalResultsToShowInStatus)
+        {
+            ss << TotalResultsTooHighStatus;
+        }
+        else
+        {
+            ss << totalMatches;
+        }
+
+        return ss.str().data();

```

Comment:
nit: Don't use `.data()` to create a `hstring`. Just construct it explicitly with `winrt::hstring{ ss.str() }`. But I would still prefer if this used `fmt` instead, for consistency and comfort. 😅 It seems like this can be done via
```cpp
fmt::format(FMT_COMPILE(L"{}/{}", arg1, arg2);
```
where arg1/2 are the results of the two if conditions.